tests/collections/common/test_lhotse_multimodal_dataloading.py

```diff
@@ -0,0 +1,214 @@
+    assert t.cut.load_audio().shape == (1, 91680)
+    t = ex.turns[2]
+    assert isinstance(t, TextTurn)
+    assert t.role == "assistant"
+    assert t.value == "I'm glad to assist you with your request. Here's a summary:"
+    t = ex.turns[3]
+    assert isinstance(t, AudioTurn)
+    assert t.role == "assistant"
+    assert t.audio_locator_tag == "[audio]"
+    assert t.cut.duration == 7.11
+    assert t.cut.load_audio().shape == (1, 113760)
+    t = ex.turns[4]
+    assert isinstance(t, TextTurn)
+    assert t.role == "user"
+    assert t.value == "Can you further shorten it?"
+    t = ex.turns[5]
+    assert isinstance(t, TextTurn)
+    assert t.role == "assistant"
+    assert t.value == "Of course!"
+    for key in ("input_ids", "context_ids", "answer_ids", "mask"):
+        assert getattr(ex, key) is None  # not tokenized/prompted
+
+
+@pytest.fixture
+def tokenizer(tmp_path_factory, multimodal_conversations_path):
+    tmpdir = tmp_path_factory.mktemp("multi_convo_tokenizer")
+    text_path = tmpdir / "text.txt"
+    text_path.write_text(
+        "\n".join(
+            turn["value"]
+            for item in lhotse.serialization.load_jsonl(multimodal_conversations_path)
+            for turn in item["conversations"]
+        )
+    )
+    create_spt_model(
+        text_path,
+        vocab_size=128,
+        sample_size=-1,
+        do_lower_case=False,
+        output_dir=str(tmpdir),
+        bos=True,
+        eos=True,
+        user_defined_symbols=["[INST]", "[/INST]", "<<SYS>>", "<</SYS>>", "[audio]"],
+    )
+    return SentencePieceTokenizer(str(tmpdir / "tokenizer.model"))
+
+
+def test_multimodal_conversation_input_with_prompt(multimodal_conversations_path, tokenizer):
+
+    config = OmegaConf.create(
+        {
+            "input_cfg": [
+                {
+                    "type": "multimodal_conversation",
+                    "manifest_filepath": multimodal_conversations_path,
+                    "audio_locator_tag": "[audio]",
+                },
+            ],
+            "prompt_format": "llama2",
+            "force_finite": True,
+            "shuffle": True,
+            "num_workers": 0,
+            "batch_size": 1,
+            "seed": 0,
+            "shard_seed": 0,
+        }
+    )
+
+    dl = get_lhotse_dataloader_from_config(
+        config=config, global_rank=0, world_size=1, dataset=Identity(), tokenizer=tokenizer
+    )
+    batches = [batch for batch in dl]
+    assert len(batches) == 1
+
+    b = batches[0]
+    assert isinstance(b, lhotse.CutSet)
+    assert len(b) == 1
+    ex = b[0]
+    assert isinstance(ex, NeMoMultimodalConversation)
+
+    assert torch.is_tensor(ex.input_ids)
+    assert ex.input_ids.shape == (111,)
+    assert (
+        tokenizer.ids_to_text(ex.input_ids)
+        == "[INST] Can you help summarize the following? [/INST] [INST] [audio] [/INST] I'm glad to assist you with your request. Here's a summary: [audio] [INST] Can you further shorten it? [/INST] Of course!"
```

Comment:
No, model responses are not wrapped by [INST]. Actually I think we may need to collapse multiple turns from the same user into a single turn so that both text and audio are wrapped by a single [INST] in the user turn. Let me make that change.

Reply:
See now - there is no `[/INST] [INST]` between the first user turn and the first user audio